### ISSUE TITLE
Logging in to ID-porten with different users from the same device results in system error

### ISSUE THREAD

**Issue by sindremb** [folkehelseinstituttet/Fhi.Smittestopp.Verification#17]
## Repro steps
1. Open smittestopp app on device X and perform verification by logging in as user A
1. Restart verification from device X and this time log in as a different user B

## Expected result
Verification for user B is completed as expected (result depending on infection status)

## Actual result
After providing credentials for user B, the ID-porten "Systemfeil"-page is shown.

## Addition info
We currently force reauthentication through ID-porten for each authorization-request by providing the request parametere prompt=login. This does seem to work to always force the login UI, but it appears there are some conflicts with the existing session from logging in as user A when you try to then log in as user B.

**Comment by sindremb** [comment]
This issue seems to be a limitation in how ID-porten interprets the prompt=login parameter. For ID-porten they expect the same user that is already authenticated (if any) to perform a reauthentication. If a different user logs in when there is a valid existing session for an other user, ID-porten will treat this as an error (since the reauthenticated user was not the expected one)

**Comment by martstr** [comment]
I would almost view this as a feature. Using two different logins from the same app instance could indicate some sort of attack, such as selling TEKs. (Although, I guess, the most common reason would be someone messing things up a bit.)

**Comment by sindremb** [comment]
@martstr Yes, I do believe this is an intended security feature. But since it keeps getting reported as a bug during testing, I have decided to keep track of it here regardless.

Unless ID-porten adds an option to their authorization where you could request a new login regardless of any existing sessions, I think the only way to solve this issue is to introduce a "Log out" feature somewhere. However, this would require us to either have a button that would be quite confusing to most users, or that we introduce a lot more state in our solution to only to be able present this button to the relevant users. Since this really is just a case that appear during testing, I guess this should just be labled as a "won't fix".
